File: suites/osprey-compute-kernels/reference/cases/pascal/pascal.c
/* Binomial coefficient via naive (un-memoised) Pascal recursion C(n,k)=C(n-1,k-1)+C(n-1,k). */
#include <stdio.h>
#include <stdint.h>

static int64_t binom(int64_t n, int64_t k) {
    if (k == 0) {
        return 1;
    } else if (k == n) {
        return 1;
    } else {
        return binom(n - 1, k - 1) + binom(n - 1, k);
    }
}

int main(void) {
    printf("%lld\n", (long long)binom(27, 13));
    return 0;
}

File: suites/osprey-compute-kernels/reference/cases/pascal/pascal.rs
// Binomial coefficient via naive (un-memoised) Pascal recursion C(n,k)=C(n-1,k-1)+C(n-1,k).
fn binom(n: i64, k: i64) -> i64 {
    if k == 0 {
        1
    } else if k == n {
        1
    } else {
        binom(n - 1, k - 1) + binom(n - 1, k)
    }
}

fn main() {
    println!("{}", binom(27, 13));
}
